db/db_local/backup.go

```diff
@@ -381,6 +382,42 @@ func getTableOfContentsFromBackup(ctx context.Context) ([]string, error) {
 	return lines, err
 }
 
+// retainBackup creates a text dump of the backup binary and saves both in the $STEAMPIPE_INSTALL_DIR/backups directory
+// the backups are saved as:
+// 		binary: 'database-yyyy-MM-dd-hh-mm-ss.dump'
+//		text:   'database-yyyy-MM-dd-hh-mm-ss.sql'
+func retainBackup(ctx context.Context) error {
+	now := time.Now()
+	backupBaseFileName := fmt.Sprintf(
+		"database-%s",
+		now.Format("2006-01-02-15-04-05"),
+	)
+	binaryBackupRetentionFileName := fmt.Sprintf("%s.dump", backupBaseFileName)
+	textBackupRetentionFileName := fmt.Sprintf("%s.sql", backupBaseFileName)
+
+	backupDir := filepaths.EnsureBackupsDir()
+	binaryBackupFilePath := filepath.Join(backupDir, binaryBackupRetentionFileName)
+	textBackupFilePath := filepath.Join(backupDir, textBackupRetentionFileName)
+
+	log.Println("[TRACE] moving database back up to", binaryBackupFilePath)
+	if err := os.Rename(databaseBackupFilePath(), binaryBackupFilePath); err != nil {
```

Comment:
is it worth doing an `io.Copy` instead of `os.Rename` here? `os.Rename` cannot work across partitions.

Reply:
may be worth it yes